gtest/src/program.rs

```diff
@@ -884,6 +884,24 @@ mod tests {
     use gear_core::ids::ActorId;
     use gear_core_errors::{ErrorReplyReason, ReplyCode, SimpleExecutionError};
 
+    #[test]
+    fn test_handle_signal() {
+        let sys = System::new();
+        sys.init_logger();
+
+        let user_id = 42;
+
+        let prog = Program::from_binary_with_id(&sys, 137, signal_in_handle::WASM_BINARY);
+
+        let run_result = prog.send_bytes(user_id, b"Does not matter");
+
+        assert!(!run_result.main_failed());
+
+        let run_result = prog.send(user_id, String::from("should panic"));
+        log::debug!("{:?}", run_result);
+        run_result.assert_panicked_with("Gotcha!");
```

Comment:
Your test program actually sends a message `handle` from program to user, which must go to mailbox.
I'd adjust the test:
1. include some payload to the message (on the contract)
2. check here that there's a message in the mailbox with the same payload. 

Currently test doesn't actually check that signals work on gtest.